compiler/rustc_resolve/src/ident.rs

```diff
@@ -952,6 +954,14 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             if !self.is_accessible_from(import_vis, parent_scope.module) {
                 continue;
             }
+            if let Some(ignored) = ignore_binding &&
+                let NameBindingKind::Import { import, .. } = ignored.kind &&
+                ptr::eq(import, &**single_import) {
+                // Ignore not just the binding itself, but if it has a shadowed_glob,
+                // ignore that, too, because this loop is supposed to only process
+                // named imports.
```

Comment:
Ah, yes, that's probably the culprit, if we have an ignored binding then we should ignore the `single_import` associated with it as well.

Reply:
The whole setup can also be simplified by filtering away the ignored binding right away - https://github.com/petrochenkov/rust/commit/b19701dd29bd7b0b2ab024f2b7abefbf054e9b95?w=1